source/agora/node/Network.d

```diff
@@ -237,4 +237,22 @@ class Network
     {
         return new RestInterfaceClient!API(address);
     }
+
+
+    /***************************************************************************
+
+        Sends the message to all the listeners.
+
+        Params:
+          msg = the message to send
+
+    ***************************************************************************/
+    
+    public void sendMessage(Hash msg) @safe
+    {
+        foreach (ref const PublicKey pk, ref RemoteNode node; this.peers)
```

Comment:
In D, you have multiple style for `foreach`. Since there's a lot of type inference, you could just replace this with:
```suggestion
        foreach (ref node; this.peers)
```

Reply:
You can also simply do `this.peers.each!(n => n.sendMessage(msg));`
https://dlang.org/phobos/std_algorithm_iteration.html#.each

Reply:
@Geod24
Check Please ~